crates/load-cargo/src/lib.rs

```diff
@@ -265,6 +265,36 @@ impl SourceRootConfig {
             })
             .collect()
     }
+
+    /// Maps local source roots to their parent source roots by bytewise comparing of root paths .
+    /// If a source root doesn't have a parent then its parent is declared as None.
+    pub fn source_root_parent_map(&self) -> FxHashMap<SourceRootId, Option<SourceRootId>> {
```

Comment:
Is the `None` case useful at all? We can just not insert an entry for the `None` case otherwise

Reply:
No it is not useful. Sorry!